src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -1703,6 +1707,20 @@ export class ChatModel extends Disposable implements IChatModel {
 			return request?.response?.isInProgress.read(r) ?? false;
 		});
 
+		// Retain a reference to itself when a request is in progress, so the ChatModel stays alive in the background
+		// only while running a request. TODO also keep it alive for 5min or so so we don't have to dispose/restore too often?
```

Comment:
That's clever 😛

Reply:
This gives me dejavu somehow, very clever!